src/otx/algorithms/detection/task.py

```diff
@@ -83,9 +83,12 @@ def __init__(self, task_environment: TaskEnvironment, output_path: Optional[str]
         )
         self._anchors: Dict[str, int] = {}
 
-        if hasattr(self._hyperparams, "postprocessing"):
-            if hasattr(self._hyperparams.postprocessing, "confidence_threshold"):
-                self.confidence_threshold = self._hyperparams.postprocessing.confidence_threshold
+        if (
+            not self._hyperparams.postprocessing.result_based_confidence_threshold
+            and hasattr(self._hyperparams, "postprocessing")
```

Comment:
Do we need check both? Looks contradictory.

Reply:
[5e71721](https://github.com/openvinotoolkit/training_extensions/pull/2388/commits/5e71721b8f0b492dc6d77ea8ae559bed3d5af3f2) changes it